packages/dart/noports_core/lib/src/srv/srv_impl.dart

```diff
@@ -406,7 +406,11 @@ class SrvImplDart implements Srv<SocketConnector> {
   @override
   Future<SocketConnector> run() async {
     try {
-      var hosts = await InternetAddress.lookup(streamingHost);
+      var relayAddresses = await InternetAddress.lookup(streamingHost);
+      if (relayAddresses.isEmpty) {
+        throw Exception('Cannot resolve relay host $streamingHost');
+      }
+      InternetAddress relayAddress = relayAddresses[0];
```

Comment:
refactor: get the relayAddress at the start, and pass it down through the various functions